flexmeasures/data/tests/test_train_predict_pipeline.py

```diff
@@ -203,6 +203,31 @@ def test_train_predict_pipeline(
             source
         ), "string representation of the Forecaster (DataSource) should mention the used model"
 
+        assert isinstance(pipeline_returns, list), "pipeline should return a list"
+        assert all(
+            isinstance(item, dict) for item in pipeline_returns
+        ), "each item should be a dict"
+        for index, pipeline_return in enumerate(pipeline_returns):
+            if not dg_params["as_job"]:
+                assert {"data", "sensor"}.issubset(
+                    pipeline_return.keys()
+                ), "returned dict should have data and sensor keys"
+                assert (
+                    pipeline_return["sensor"].id == dg_params["sensor_to_save"].id
+                ), "returned sensor should match sensor that forecasts will be saved into"
+                try:
+                    pd.testing.assert_frame_equal(
+                        forecasts.sort_index(),
+                        pipeline_return["data"].sort_index(),
+                    )
+                except AssertionError as e:
+                    raise AssertionError(
+                        f"returned data should match stored forecasts: {e}"
+                    )
+            else:
+                assert f"job-{index}" in pipeline_return
+                assert isinstance(pipeline_return[f"job-{index}"], str)
```

Comment:
Instead of testing for a string uuid, here you could test if a `Job` object with the given uuid can be found in the forecasting queue, and that it's in the *finished* registry.